esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2133,9 +2243,9 @@ bool CFileSprayEx::onSprayVariable(IEspContext &context, IEspSprayVariable &req,
             gName.append(destNodeGroup);
 
         MemoryBuffer& srcxml = (MemoryBuffer&)req.getSrcxml();
-        StringBuffer sourcePlaneReq, sourceIPReq, sourcePathReq;
-        readAndCheckSpraySourceReq(srcxml, req.getSourceIP(), req.getSourcePath(), req.getSourcePlane(), sourcePlaneReq, sourceIPReq, sourcePathReq);
-        const char* srcfile = sourcePathReq.str();
+        StringBuffer sourcePlaneReq, sourceHostReq, sourcePathReq;
+        if (srcxml.length() == 0)
+            readAndCheckSpraySourceReq(context, req.getSourceIP(), req.getSourcePath(), req.getSourcePlane(), sourcePlaneReq, sourceHostReq, sourcePathReq);
```

Comment:
not for now: there's a lot in common with onSprayFixed vs onSprayVariable, but differences that I can't immediately see why are there, e.g. onSprayVariable has:
```
        StringBuffer gName, ipAddr;
        const char *pTr = strchr(destNodeGroup, ' ');
        if (pTr)
        {
            gName.append(pTr - destNodeGroup, destNodeGroup);
            ipAddr.append(pTr+1);
        }
        else
            gName.append(destNodeGroup);
```

They should be as similar as possible, preferably using common code wherever possible.
Not for now, but could the ecm EspSpray*'s be commoned up and the *Fixed vs *Variable definitions be derived from EspSprayCommon? Such that these methods wouldn't have to duplicate as much?